### PR TITLE
chore: update swagger document

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] update swagger document

### CODE REVIEW

File: pkg/handler/audit/audit.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ func New(store *store.Store, repo store.DBRepo, service *service.Service, logger
 // @Tags Audit
 // @Accept  json
 // @Produce  json
-// @Param Authorization header string true "jwt token"
-// @Success 200 {object} view.MessageResponse
+// @Security BearerAuth
+// @Success 200 {object} MessageResponse
```
Comment (hieuphq) [inline pkg/handler/audit/audit.go:53]: are u missing // @id

File: pkg/handler/bankaccount/bank_account.go [hunk 2]
```diff
@@ -34,11 +34,12 @@ func New(store *store.Store, repo store.DBRepo, service *service.Service, logger
 // List godoc
 // @Summary Get all bank accounts
 // @Description Get all bank accounts
+// @id listBankAccounts
 // @Tags Bank
 // @Accept  json
 // @Produce  json
-// @Success 200 {object} []view.ListBankAccountResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Success 200 {object} ListBankAccountResponse
```
Comment (hieuphq) [inline pkg/handler/bankaccount/bank_account.go:41]: @namnhce is this right?

File: pkg/handler/brainerylogs/brainery_log.go [hunk 3]
```diff
@@ -50,13 +50,14 @@ const (
...
+// @id createBraineryLogs
 // @Tags Project
 // @Accept  json
 // @Produce  json
-// @Param Authorization header string true "jwt token"
-// @Success 200 {object} view.MessageResponse
-// @Failure 400 {object} view.ErrorResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Security BearerAuth
+// @Success 200 {object} MessageResponse
```
Comment (hieuphq) [inline pkg/handler/brainerylogs/brainery_log.go:58]: missing body

File: pkg/handler/brainerylogs/brainery_log.go [hunk 4]
```diff
@@ -50,13 +50,14 @@ const (
 // Create godoc
 // @Summary Create brainery logs
 // @Description Create brainery logs
+// @id createBraineryLogs
```
Comment (hieuphq) [inline pkg/handler/brainerylogs/brainery_log.go:53]:
```suggestion
// @id createBraineryLog
```

File: pkg/handler/brainerylogs/brainery_log.go [hunk 5]
```diff
@@ -103,15 +104,16 @@ func (h *handler) Create(c *gin.Context) {
...
 // @Produce  json
-// @Param Authorization header string true "jwt token"
+// @Security BearerAuth
 // @Param view query string false "Time view"
 // @Param date query string false "Date"
-// @Success 200 {object} view.BraineryMetric
-// @Failure 400 {object} view.ErrorResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Success 200 {object} BraineryMetricResponse
+// @Failure 400 {object} ErrorResponse
```
Comment (hieuphq) [inline pkg/handler/brainerylogs/brainery_log.go:115]: missing query param

File: pkg/handler/client/client.go [hunk 6]
```diff
@@ -105,11 +106,11 @@ func (h *handler) List(c *gin.Context) {
 // @Tags Client
 // @Accept  json
 // @Produce  json
-// @Param Authorization header string true "jwt token"
-// @Success 200 {object} view.GetDetailClientResponse
-// @Failure 400 {object} view.ErrorResponse
-// @Failure 404 {object} view.ErrorResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Security BearerAuth
```
Comment (hieuphq) [inline pkg/handler/client/client.go:109]: mising id param

File: pkg/handler/client/client.go [hunk 7]
```diff
@@ -144,12 +145,12 @@ func (h *handler) Detail(c *gin.Context) {
 // @Tags Client
 // @Accept  json
 // @Produce  json
```
Comment (hieuphq) [inline pkg/handler/client/client.go:147]: missing // @id

File: pkg/handler/client/client.go [hunk 8]
```diff
@@ -186,10 +187,10 @@ func (h *handler) Update(c *gin.Context) {
...
```
Comment (hieuphq) [inline pkg/handler/client/client.go:189]: missing // @id

File: pkg/handler/client/client.go [hunk 9]
```diff
@@ -186,10 +187,10 @@ func (h *handler) Update(c *gin.Context) {
...
-// @Success 200 {object} view.MessageResponse
-// @Failure 400 {object} view.ErrorResponse
-// @Failure 404 {object} view.ErrorResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Success 200 {object} MessageResponse
```
Comment (hieuphq) [inline pkg/handler/client/client.go:190]: missing id param

File: pkg/handler/dashboard/dashboard.go [hunk 10]
```diff
@@ -46,9 +46,9 @@ func New(store *store.Store, repo store.DBRepo, service *service.Service, logger
 // @Tags Dashboard
 // @Accept json
 // @Produce json
```
Comment (hieuphq) [inline pkg/handler/dashboard/dashboard.go:48]: missing // @id

File: pkg/handler/dashboard/dashboard.go [hunk 11]
```diff
@@ -46,9 +46,9 @@ func New(store *store.Store, repo store.DBRepo, service *service.Service, logger
...
-// @Param Authorization header string true "jwt token"
+// @Security BearerAuth
 // @Success 200 {object} view.ProjectSizeResponse
```
Comment (hieuphq) [inline pkg/handler/dashboard/dashboard.go:50]:
```suggestion
// @Success 200 {object} ProjectSizeResponse
```

File: pkg/handler/dashboard/dashboard.go [hunk 12]
```diff
@@ -72,12 +72,12 @@ func (h *handler) GetProjectSizes(c *gin.Context) {
...
 // @Produce json
-// @Param Authorization header string true "jwt token"
+// @Security BearerAuth
 // @Param projectID   query  string false  "Project ID"
 // @Success 200 {object} view.WorkSurveyResponse
-// @Failure 400 {object} view.ErrorResponse
-// @Failure 404 {object} view.ErrorResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Failure 400 {object} ErrorResponse
+// @Failure 404 {object} ErrorResponse
```
Comment (hieuphq) [inline pkg/handler/dashboard/dashboard.go:79]: missing query params

File: pkg/handler/dashboard/dashboard.go [hunk 13]
```diff
@@ -72,12 +72,12 @@ func (h *handler) GetProjectSizes(c *gin.Context) {
 // @Tags Dashboard
 // @Accept json
 // @Produce json
-// @Param Authorization header string true "jwt token"
+// @Security BearerAuth
```
Comment (hieuphq) [inline pkg/handler/dashboard/dashboard.go:75]: missing // @id

File: pkg/handler/dashboard/dashboard.go [hunk 14]
```diff
@@ -141,12 +141,12 @@ func (h *handler) GetWorkSurveys(c *gin.Context) {
...
```
Comment (hieuphq) [inline pkg/handler/dashboard/dashboard.go:144]: mising // @id

File: pkg/handler/dashboard/dashboard.go [hunk 15]
```diff
@@ -141,12 +141,12 @@ func (h *handler) GetWorkSurveys(c *gin.Context) {
...
 // @Param projectID   query  string false  "Project ID"
 // @Success 200 {object} view.ActionItemReportResponse
-// @Failure 400 {object} view.ErrorResponse
-// @Failure 404 {object} view.ErrorResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Failure 400 {object} ErrorResponse
+// @Failure 404 {object} ErrorResponse
```
Comment (hieuphq) [inline pkg/handler/dashboard/dashboard.go:148]: missing query param

**Review by hieuphq — CHANGES_REQUESTED** [review]

File: pkg/handler/dashboard/dashboard.go (hunk 11 above)
Comment (datphamcode295) [reply pkg/handler/dashboard/dashboard.go:50]: chỗ này nó là của project chưa chỉnh tới mà anh em mới chỉnh chung cái jwt thôi

File: pkg/handler/dashboard/dashboard.go (hunk 12 above)
Comment (datphamcode295) [reply pkg/handler/dashboard/dashboard.go:79]: this PR just fix to the "client" so we should ignore d.., e..., f... folder

File: pkg/handler/payroll/payroll.go [hunk 16]
```diff
@@ -63,7 +63,7 @@ func New(controller *controller.Controller, store *store.Store, repo store.DBRep
 // @Accept  json
 // @Produce  json
 // @Success 200 {object} []model.Payroll
-// @Failure 400 {object} view.ErrorResponse
```
[the remark this answers is not in the document]
Comment (datphamcode295) [reply pkg/handler/payroll/payroll.go:66]: updated

File: pkg/handler/audit/audit.go [hunk 17]
```diff
@@ -46,11 +46,12 @@ func New(store *store.Store, repo store.DBRepo, service *service.Service, logger
 // Sync godoc
 // @Summary Sync audit info from Notion to database
 // @Description Sync audit info from Notion to database
+// @id SyncAudit
```
Comment (hieuphq) [inline pkg/handler/audit/audit.go:49]: quỳ

File: pkg/handler/client/client.go [hunk 18]
```diff
@@ -42,13 +42,14 @@ func New(controller *controller.Controller, store *store.Store, repo store.DBRep
...
+// @id createClient
 // @Tags Client
 // @Accept  json
 // @Produce  json
-// @Param Authorization header string true "jwt token"
-// @Param Body body request.CreateClientInput true "Body"
-// @Success 200 {object} view.CreateClientResponse
-// @Failure 500 {object} view.ErrorResponse
+// @Security BearerAuth
+// @Param Body body CreateClientInput true "Body"
```
Comment (hieuphq) [inline pkg/handler/client/client.go:50]:
```suggestion
// @Param Body body CreateClientRequest true "Body"
```

File: pkg/handler/client/client.go [hunk 19]
```diff
@@ -70,18 +71,19 @@ func (h *handler) Create(c *gin.Context) {
...
 	}
 
-	c.JSON(http.StatusOK, view.CreateResponse[any](client, nil, nil, nil, ""))
+	c.JSON(http.StatusOK, view.CreateResponse[any](view.ToClient(client), nil, nil, nil, ""))
 }
 
 // List godoc
 // @Summary Get all clients
 // @Description Get all clients
+// @id listClients
```
Comment (hieuphq) [inline pkg/handler/client/client.go:80]: so confuse. Sth we using prefix `get` sth nothing

File: pkg/handler/client/client.go (hunk 9 above)
Comment (hieuphq) [reply pkg/handler/client/client.go:195]: missing missing missing ID in query param >.<

File: pkg/handler/client/client.go [hunk 20]
```diff
@@ -217,12 +223,13 @@ func (h *handler) Delete(c *gin.Context) {
 // PublicList godoc
 // @Summary Get all clients
 // @Description Get all clients
+// @id listPublicClients
```
Comment (hieuphq) [inline pkg/handler/client/client.go:226]: confusing with the inconsistent id

**Review by hieuphq — CHANGES_REQUESTED** [review]

File: pkg/view/response.go [hunk 21]
```diff
@@ -4,10 +4,25 @@ import (
...
+		return nil
+	}
+
+	var deletedAt *time.Time
+	if value.Valid {
+		deletedAt = &value.Time
+	}
+
+	return deletedAt
+}
```
[the remark this answers is not in the document]
Comment (datphamcode295) [reply pkg/view/response.go:24]: removed

File: pkg/handler/client/client.go [hunk 22]
```diff
@@ -217,12 +225,13 @@ func (h *handler) Delete(c *gin.Context) {
 // PublicList godoc
 // @Summary Get all clients
 // @Description Get all clients
+// @id getListPublicClients
```
Comment (hieuphq) [inline pkg/handler/client/client.go:228]:
```suggestion
// @id getAllPublicClients
```

**Review by hieuphq — CHANGES_REQUESTED** [review]

File: pkg/view/client.go [hunk 23]
```diff
@@ -2,50 +2,86 @@ package view
...
+	Lat                string
+	Long               string
+	CompanySize        string
+	SolutionType       string
+
+	Contacts []ClientContact
+	Projects []Project
+} // @name Client
+
+func ToClients(clients []*model.Client) []Client {
```
[the remark this answers is not in the document]
Comment (hieuphq) [reply pkg/view/client.go:45]: @namnhce @datphamcode295

File: pkg/handler/client/client.go (hunk 22 above)
Comment (datphamcode295) [reply pkg/handler/client/client.go]: updated

File: pkg/handler/bankaccount/bank_account.go (hunk 2 above)
Comment (hieuphq) [reply pkg/handler/bankaccount/bank_account.go:41]: I mean why we have the different between 2 of them: array and object

File: pkg/view/client.go (hunk 23 above)
[the remark this answers is not in the document]
Comment (datphamcode295) [reply pkg/view/client.go:45]: nooooo

File: pkg/handler/bankaccount/bank_account.go (hunk 2 above)
Comment (datphamcode295) [reply pkg/handler/bankaccount/bank_account.go:41]: value []view.ListBankAccountResponse was wrong, the response of the function is {Data []BankAccount `json:"data"`} = ListBankAccountResponse so I updated it to new model ListBankAccountResponse

File: pkg/view/client.go (hunk 23 above)
[the remark this answers is not in the document]
Comment (hieuphq) [reply pkg/view/client.go:45]: why u don't answer my question from begining :D reject again

**Review by hieuphq — APPROVED** [review]

**Pull request merged**